### ISSUE TITLE
How to implements L1 and L2 regularization?

### ISSUE THREAD

**Issue by richardhahahaha** [torch/optim#59]

**Comment by soumith** [comment]
one of them is here: https://github.com/torch/optim/blob/master/sgd.lua#L11
you can take a hint on how to implement the other.

**Comment by amirkhango** [comment]
But this is for L2 regularization, how to do L1 regularization?

**Comment by nagadomi** [comment]
As far as I know, [FOBOS](http://www.jmlr.org/papers/volume10/duchi09a/duchi09a.pdf) is one of implementation technique for training L1 regularized model with gradient descent.
